### PR TITLE
Prepare for httpProbes

### PR DESCRIPTION
**Pull request by alexellis**
Signed-off-by: Alex Ellis <alexellis2@gmail.com>

## Description

Prepare to switch to httpProbes and fix issues.

- httpProbe as an option is documented in the README

- In testing the update handler would not update the probe values
for deployments even if the controller was reloaded with new
config. This has been updated in the update handler.

- New method broken out for makeProbes to be used from
deploy/update.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

Fixes: #413

## How Has This Been Tested?

* Updated unit tests

* Scenario 1 - not updating the probes

Deployed faas-netes with current image and httpProbe=false, deployed cows. Then I updated the faas-netes deployment for httpProbe=true and redeployed cows, but the probes were not changed. With a new image from this PR this scenario was resolved.

* Scenario 2 - exec / http probes both work

In this example I tried toggling httpProbe on/off and was able to see it reflect in the function and the function was healthy. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: contrib/deploy.sh [hunk 1]
```diff
@@ -5,8 +5,17 @@ export KUBECONFIG="$(kind get kubeconfig-path)"
 echo "Applying namespaces"
 kubectl apply -f ./namespaces.yml
 
-echo $(head -c 16 /dev/urandom| sha256sum | cut -d " " -f 1) > ./password.txt # Store password in password.txt file
-PASSWORD=$(cat ./password.txt)
+sha_cmd="sha256sum"
```
Comment (alexellis) [inline contrib/deploy.sh]: @zeerorg FYI this broke on MacOS so I've put a fix in that @rgee0 added in the faas-cli.

**Comment by alexellis** [comment]
@stefanprodan @LucasRoesler we'll need to apply this same or a similar patch to the openfaas-operator.

**Comment by alexellis** [comment]
> This seems like a good opportunity to add a test for the probes configuration methods. Especially since we plan to make it mor complex soon.

Which method?

Make it more complex how?

**Comment by LucasRoesler** [comment]
I am going to take a break from my computer tonight, because I can't seem to keep my sentences straight

**Comment by alexellis** [comment]
Thanks for the suggestion. I've added a unit test for both cases (exec/httpProbe) and would be happy to add more later.

Have a good evening 👋

File: contrib/deploy.sh (hunk 1 above)
Comment (zeerorg) [reply contrib/deploy.sh]: I should've taken more care with sha256 utility, thanks for fixing this.

**Pull request merged**